streams/service/src/main/java/com/hortonworks/streamline/streams/service/TopologyTestRunResource.java

```diff
@@ -736,4 +774,114 @@ public Long getTimestamp() {
         }
     }
 
+    public static class EventInformation {
+        private long timestamp;
+        private String componentName;
+        private String eventId;
+        private Set<String> rootIds;
+        private Set<String> parentIds;
+        private Map<String, Object> fieldsAndValues;
+
+        // jackson
+        public EventInformation() {
+        }
+
+        public EventInformation(long timestamp, String componentName, String eventId, Set<String> rootIds,
+                                Set<String> parentIds, Map<String, Object> fieldsAndValues) {
+            this.timestamp = timestamp;
+            this.componentName = componentName;
+            this.eventId = eventId;
+            this.rootIds = rootIds;
+            this.parentIds = parentIds;
+            this.fieldsAndValues = fieldsAndValues;
+        }
+
+        public long getTimestamp() {
+            return timestamp;
+        }
+
+        public String getComponentName() {
+            return componentName;
+        }
+
+        public String getEventId() {
+            return eventId;
+        }
+
+        public Set<String> getRootIds() {
+            return rootIds;
+        }
+
+        public Set<String> getParentIds() {
+            return parentIds;
+        }
+
+        public Map<String, Object> getFieldsAndValues() {
+            return fieldsAndValues;
+        }
+    }
+
+    private Stream<EventInformation> loadEventLogFile(Long topologyId, Long historyId) throws IOException {
+        File eventLogFile = getEventLogFile(topologyId, historyId);
+
+        List<String> lines = FileUtils.readLines(eventLogFile, ENCODING_UTF_8);
+        return lines.stream().map(line -> {
+            try {
+                return objectMapper.readValue(line, new TypeReference<EventInformation>() {});
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    public static class EventInformationTreeNode {
```

Comment:
There're some methods which handles event log file, and also handles event trees. Maybe we could extract a new class for that.

Reply:
Can you move this and EventInformation classes as independent classes in some common package so that it can be re-used?

Reply:
OK. I'll do it.